code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LdapIdentitySearchProvider.java

```diff
@@ -195,19 +199,39 @@ private Attributes userRecord(LdapContext context, String scope, String query) {
     }
 
     private Set<Identity> getIdentities(Attributes userAttributes) {
-        Set<Identity> identities = new HashSet<>();
-        Attribute memberOf = userAttributes.get(LdapConstants.MEMBER_OF);
+        final Set<Identity> identities = new HashSet<>();
+        final Attribute memberOf = userAttributes.get(LdapConstants.MEMBER_OF);
         try {
             if (!isType(userAttributes, LdapConstants.USER_OBJECT_CLASS.get()))
             {
                 return identities;
             }
+            Collection<Callable<Identity>> groupsToGet = new ArrayList<>();
             identities.add(attributesToIdentity(userAttributes));
             if (memberOf != null) {// null if this user belongs to no group at all
                 for (int i = 0; i < memberOf.size(); i++) {
-                    identities.add(getObject(memberOf.get(i).toString(), LdapConstants.GROUP_SCOPE));
+                    final int finalI = i;
+                    groupsToGet.add(new Callable<Identity>() {
+                        @Override
+                        public Identity call() throws Exception {
+                            try {
+                                Identity identity = getObject(memberOf.get(finalI).toString(), LdapConstants.GROUP_SCOPE);
```

Comment:
Only if all of a sudden the ldap entry lost the attribute i am getting. So yes. but highly unlikely.